From c68373de3d9552f0198ea9a07c9b091fa89406cb Mon Sep 17 00:00:00 2001
From: Konrad Kleine <kkleine@redhat.com>
Date: Thu, 12 Dec 2024 11:22:33 +0100
Subject: [PATCH] Properly install tsl::robin_map's interface sources

Before, the include files from `ext/robin_map/include/tsl` were installed
manually. This is when I noticed that one file from `robin_map` was
missing, namely `robin_set.h`.

When I found the place in which nanobind installs the headers I looked
up the `CMakeLists.txt` from `robin_map`. One can safely add it with
`add_subdirectory()` because it doesn't do much when added as a
subproject. The `README.txt` from `robin_map` says:

> If you use CMake, you can also use the `tsl::robin_map` exported target
> from the `CMakeLists.txt`.

That's when I figured we might want to make this a bit more future proof
and automatically get the header files from `robin_map`.
---
 CMakeLists.txt | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f94783e..1507173 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -96,10 +96,9 @@ if(NB_CREATE_INSTALL_RULES AND NOT CMAKE_SKIP_INSTALL_RULES)
   )
 
   if(NB_USE_SUBMODULE_DEPS)
+    add_subdirectory(ext/robin_map)
     install(
-      FILES ext/robin_map/include/tsl/robin_map.h
-            ext/robin_map/include/tsl/robin_hash.h
-            ext/robin_map/include/tsl/robin_growth_policy.h
+      FILES  $<TARGET_PROPERTY:tsl::robin_map,INTERFACE_SOURCES>
       DESTINATION "${NB_INSTALL_EXT_DIR}/robin_map/include/tsl"
     )
   endif()
-- 
2.47.1

